archiver.go

```diff
@@ -200,12 +199,21 @@ func (a *Archiver) do(log log15.Logger, j *Job) (err error) {
 	return nil
 }
 
-func (a *Archiver) canProcessRepository(repo *model.Repository) error {
+func (a *Archiver) canProcessRepository(repo *model.Repository, now *time.Time) (err error) {
+	defer func() {
+		if err != nil {
```

Comment:
why is this logic in a defer instead of just after the `if` checking the status?

Reply:
Because if in the future we have more wrong statuses we can handle all of them just with a `switch` and setting err for each case. Should I change it for the current only one case?